src/main/scala/symsim/concrete/ConcreteExactRL.scala

```diff
@@ -33,3 +33,24 @@ trait ConcreteExactRL[State, ObservableState, Action]
   /** Convert the matrix Q after training into a Policy map. */
   def qToPolicy (q: Q) (using Ordering[Double]): Policy =
     q.states.map { s => (s, vf.bestAction (q) (s)) }.to (Map)
+
+  def policyEval (policies: List[Policy]): List[List[Double]] =
+    val initials = Randomized.repeat (agent.initialize).take (5)
+    val rL = for
+      s_t <- initials
+    yield policies.map (policy => runEval (policy, s_t, 0.0))
+    rL.toList
+
+  def runEval(policy: Policy, s_t: State, r_acc: Double): Double =
+    if agent.isFinal(s_t) then r_acc
```

Comment:
it is really not clear what is a good reward for a final state. I suggest we assume that the initial state is not final. Remove the if, just keep the else branch, and add `require(!agent.isFinal(s_t), s"Evaluation cannot start in a final state $s_t")` instead.